azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -493,6 +520,13 @@ def get_command_line(pid):
 
             return [get_command_line(pid) for pid in pid_list]
 
+        @staticmethod
+        def _check_agent_throttled_time(cgroup_metrics):
```

Comment:
This is just a bare-bones implementation. As i work on the end-to-end tests I'll adjust the logic accordingly